src/pylhe/awkward.py

```diff
@@ -10,12 +10,22 @@ def __dir__():
 
 
 def register_awkward():
-    """Register Awkward behaviors for pylhe."""
+    """This function does nothing and is merely here for backwards compatibility
+    of the pylhe API
 
-    vector.register_awkward()
-    ak.mixin_class(ak.behavior)(Particle)
-    ak.mixin_class(ak.behavior)(Event)
-    ak.mixin_class(ak.behavior)(EventInfo)
+    The registration is done automatically when pylhe is imported since awkward
+    is a required dependency of pylhe.
+    """
+    import warnings
+
+    warnings.warn(
+        """The `pylhe.register_awkward` function does nothing.
+            The registration now happens automatically when `pylhe` is imported,
+            so calls to `pylhe.register_awkward` can be safely removed.""",
+        category=DeprecationWarning,
+        stacklevel=2,
+    )
+    pass
```

Comment:
I'd prefer to use the system we have for `pyhf` of giving an explicit deprecated API warning with a removal version (2 minor releases). c.f.

* https://github.com/scikit-hep/pyhf/blob/d6208cbabb8ab727f201e6e06673f938d77b14b2/src/pyhf/exceptions/__init__.py#L181-L190
* https://github.com/scikit-hep/pyhf/blob/d6208cbabb8ab727f201e6e06673f938d77b14b2/src/pyhf/infer/intervals/__init__.py#L11-L30

No need to make an exceptions system here, I was just showing what the warning system looks like.